### DISCUSSION TITLE
Function with "load_all_pages" decorator returns generator object

### DISCUSSION THREAD

**Discussion by mudassirzr** [saleweaver/python-amazon-sp-api#294] [Q&A] [answered]
This might be a silly question to ask but I'm facing an issue with the `load_all_pages` decorator. If I do something like this:
```
@throttle_retry()
@load_all_pages()
def get_orders(**kwargs):
    return Orders().get_orders(**kwargs)
```
and then:
```
orders = get_orders(**params)
print(orders)
```
the `get_orders` returns a generator object:
```
<generator object load_all_pages.<locals>.decorator.<locals>.wrapper at 0x7fab3859dba0>
```
Ideally it should return the orders but seems something is wrong with either my code or my approach. Would appreciate any help with what could be wrong with my code and what I should do to resolve it.

Thanks

**Comment by saleweaver** [comment, marked as answer]
That's the documented behavior: https://sp-api-docs.saleweaver.com/utils/load_all_pages/

Usage is further explained here: https://sp-api-docs.saleweaver.com/quickstart/

**Reply by mudassirzr** [reply]
Thanks for the answer, hadn't seen the quickstart example, would be better if a similar code snippet is shared on `load_all_pages` decorator documentation page.

**Reply by saleweaver** [reply]
Thanks for the feedback

**Reply by saleweaver** [reply]
I've added the code example to the page: https://sp-api-docs.saleweaver.com/utils/load_all_pages/
Thanks again!

**Reply by mudassirzr** [reply]
Thanks, appreciate it!
